react/Menu/index.jsx

```diff
@@ -42,24 +43,28 @@ class Menu extends Component {
       this.close()
     }
   }
+  // Sum the scrollTop between the element and all the tree up
+  getScrollParent(element) {
+    let el = element
+    let scrollTop = 0
 
-  handleClick = item => {
-    const isDisabled = item.props.disabled
-    const itemHandler = isDisabled ? null : item.props.onSelect
-    const handler =
-      itemHandler || this.props[!isDisabled ? 'onSelect' : 'onSelectDisabled']
-    if (handler) {
-      const res = handler(item.props.data)
-      if (res !== false) {
-        this.close()
-      }
-    } else if (!isDisabled) {
-      this.close()
+    while (el && el.parentNode !== document) {
+      scrollTop += el.scrollTop
+      el = el.parentNode
     }
+    return scrollTop
   }
```

Comment:
I'm not sure but I think it can be possible to have the coordinates/scroll logic only in one component. Here, it is spread between two components and I find the responsibilities of each one hard to grasp.